scripts/lighthouse_checks.py

```diff
@@ -0,0 +1,172 @@
+RUNNING_PROCESSES = []
+GO_VERSION = '1.14.1'
+GO_PATH = os.path.join(common.OPPIA_TOOLS_DIR, 'go-%s' % GO_VERSION)
+GO_BINARY = os.path.join(GO_PATH, 'bin', 'go')
+GO_HOME_PATH = os.path.join(os.path.expanduser('~'), 'go')
+
+COMPY_BINARY = os.path.join(GO_HOME_PATH, 'bin', 'compy')
+COMPY_DOWNLOAD_PATH = os.path.join(
+    GO_HOME_PATH, 'src', 'github.com', 'barnacs', 'compy')
+MAX_WAIT_TIME_FOR_PORT_TO_OPEN_SECS = 1000
+
+
+def setup_and_install_dependencies():
+    """Run the setup and installation scripts."""
+
+    install_third_party_libs.main()
+    setup.main(args=[])
+    setup_gae.main(args=[])
+
+
+def start_google_app_engine_server():
+    """Start the Google App Engine server."""
+    app_yaml_filepath = 'app.yaml'
+
+    p = subprocess.Popen(
+        '%s %s/dev_appserver.py --host 0.0.0.0 --port %s '
+        '--clear_datastore=yes --dev_appserver_log_level=critical '
+        '--log_level=critical --skip_sdk_update_check=true %s' % (
+            common.CURRENT_PYTHON_BIN, common.GOOGLE_APP_ENGINE_HOME,
+            OPPIA_SERVER_PORT, app_yaml_filepath), shell=True)
+    RUNNING_PROCESSES.append(p)
+
+
+def download_and_install_go():
+    """Download and install GO."""
+    outfile_name = 'go-download'
+
+    if common.is_windows_os():
+        pass
+    else:
+        extension = '.tar.gz'
+
+        if common.is_mac_os():
+            go_file_name = 'go%s.darwin-amd64' % (GO_VERSION)
+        elif common.is_linux_os():
+            go_file_name = 'go%s.linux-amd64' % (GO_VERSION)
+
+        url_to_retrieve = 'https://storage.googleapis.com/golang/%s%s' % (
+            go_file_name, extension)
+
+        python_utils.url_retrieve(url_to_retrieve, outfile_name)
+
+        subprocess.check_call([
+            'tar', '-xvzf', outfile_name, '-C', common.OPPIA_TOOLS_DIR
+        ])
+
+    os.rename(
+        os.path.join(common.OPPIA_TOOLS_DIR, 'go'),
+        GO_PATH)
+    python_utils.PRINT('Installation successful')
+
+
+def download_and_install_compy():
+    """Download and install the compy proxy server."""
+
+    subprocess.check_call([GO_BINARY, 'get', 'github.com/barnacs/compy'])
+    with common.CD(COMPY_DOWNLOAD_PATH):
+        subprocess.check_call([GO_BINARY, 'install'])
+    python_utils.PRINT('Installation successful')
+
+
+def cleanup():
+    """Kill the running subprocesses and server fired in this program."""
+    dev_appserver_path = '%s/dev_appserver.py' % common.GOOGLE_APP_ENGINE_HOME
+    processes_to_kill = [
+        '.*%s.*' % re.escape(dev_appserver_path),
+        '.*%s.*' % re.escape(COMPY_BINARY)
+    ]
+
+    for p in RUNNING_PROCESSES:
+        p.kill()
+
+    for p in processes_to_kill:
+        common.kill_processes_based_on_regex(p)
+
+
+def delete_reports():
+    """"Delete the reports that are stored in the lighthouse ci folder."""
+    bash_command = 'rm -r .lighthouseci'
+    process = subprocess.Popen(bash_command.split(), stdout=subprocess.PIPE)
+    process.communicate()
```

Comment:
Delete the files by Python methods from the `os` stdlib and not by command.

Reply:
Done